src/DenseACSets.jl

```diff
@@ -464,18 +511,18 @@ end
   delete!(acs.subparts[f], part)
 
 @inline ACSetInterface.rem_part!(acs::StructACSet{S}, type::Symbol, part::Int) where {S} =
-  _rem_part!(acs, Val{S}, Val{type}, part)
+  _rem_part!(acs, Val{S}, Val{type}, part, acs.parts[type])
 
 ACSetInterface.rem_part!(acs::DynamicACSet, type::Symbol, part::Int) =
-  runtime(_rem_part!, acs, acs.schema, type, part)
+  runtime(_rem_part!, acs, acs.schema, type, part, acs.parts[type])
 
-@ct_enable function _rem_part!(acs::SimpleACSet, @ct(S), @ct(ob), part)
+@ct_enable function _rem_part!(acs::SimpleACSet, @ct(S), @ct(ob), part, ::IntParts)
```

Comment:
This should dispatch based on the strategy, not the concrete type of the parts, just like how we don't ever dispatch on column type in this file.